script.js

```diff
@@ -1,25 +1,237 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  const episodeCards = makePageForEpisodes(allEpisodes);
-  document.body.append(...episodeCards);
+const showDropDownMenu = document.querySelector('#selectShow');
+let EpisodeDropDown = document.querySelector('#selectEpisode')
+const input = document.querySelector('#input');
+const readMore = document.querySelector('#readMore');
+const home = document.querySelector('#home');
+
+let selectedShow = 1;
+let selectedEpisode = "";
+let SearchTerm = "";
+
+// fetch Api function to fetch the shows
+async function getAllShows() {
+  return await fetch("https://api.tvmaze.com/shows").then((data) => {
+    return data.json();
+  })
+}
+// fetch Api function to fetch the Episodes of selected show
+async function getAllEpisodes() {
```

Comment:
It is generally considered to be good practice to provide the external data needed by the function as an argument. For example, it would be better to pass `selectedShow` as an argument rather than relying directly on the global variable. The global variable can be modified elsewhere in the code while the function is still in the middle of running and can lead to unexpected behaviour in certain cases.

Reply:
I tried to pass `selectedShow` as argument but somehow it was't working. I will look into it again. Thank you.